common/djangoapps/student/views.py

```diff
@@ -1804,6 +1805,27 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _enroll_user_in_pending_courses(student):
+    """
+    Enroll student in any pending courses he/she may have.
+    """
+    if student:
```

Comment:
Okay. |'ll add this condition.
No this function is called in another view `activate_account`.When activation e-mail link is clicked, its activate the account and then this function `enroll_user_in_pending_courses` is called to enroll student in pending courses. That's why its return the user.